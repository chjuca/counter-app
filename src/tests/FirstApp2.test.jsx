import { render, screen } from "@testing-library/react"
import FirstApp from "../FirstApp"

describe("Test <FirstApp/>", () =>{

    const title = "Hola, soy Carlos"
    const subTitle = "soy un subtitulo"

    test('Should match snapshot', () => {

        const {container} = render(<FirstApp title={title}/>)
        expect (container).toMatchSnapshot()
    })


    test('Should match screen title', () => {

        render(<FirstApp title={title}/>)
        expect (screen.getByText(title)).toBeTruthy()
    })

    test('Should show title in h1', () => {

        render(<FirstApp title={title}/>)
        expect (screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })

    test('Should show subtitle send in props', () => {

        render(<FirstApp title={title} subTitle={subTitle}/>)
        expect (screen.getAllByText(subTitle).length).toBe(2)
    })


})