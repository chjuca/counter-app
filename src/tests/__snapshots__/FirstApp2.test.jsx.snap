// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp/> Should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Carlos
  </h1>
  <p />
  <p />
  <p />
</div>
`;
